lib/dialect/src/ConstraintsTestPass.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2024 Matteo Cenzato 
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+#include "llvm/ADT/TypeSwitch.h"
+#include "mlir/IR/MLIRContext.h"
+#include "rlc/dialect/Operations.hpp"
+#include "rlc/dialect/Passes.hpp"
+#include "rlc/dialect/Types.hpp"
+#include <iostream>
+#include <limits>
+
+namespace mlir::rlc
+{
+
+#define GEN_PASS_DEF_CONSTRAINTSTESTPASS
```

Comment:
why do we have this file if the analysis results to the module are attached in the other pass?
i guess this should be dropped?

Reply:
And it will be dropped.